lib/cacheOptimization.js

```diff
@@ -0,0 +1,84 @@
+function getAccessorsPrimitive(gltf, attributeAccessors, indicesAccessor) {
+    var meshes = gltf.meshes;
+    for (var meshId in meshes) {
+        if (meshes.hasOwnProperty(meshId)) {
+            var mesh = meshes[meshId];
+            var primitives = mesh.primitives;
+            var primitivesLength = primitives.length;
+            for (var i = 0; i < primitivesLength; i++) {
+                var primitive = primitives[i];
+                var accessors = primitive.attributes;
+
+                var equalCount = 0;
+                // Naive check to see if this primitive's attributes match
+                for (var accessorId in accessors) {
+                    if (accessors.hasOwnProperty(accessorId)) {
+                        for (var attributeAccessor in attributeAccessors) {
+                            if (attributeAccessor === accessors[accessorId]) {
+                                equalCount++;
+                            }
+                        }
+                    }
+                }
+                var primitiveIndices = primitive.indices;
+                var accessorsLength = Object.keys(attributeAccessors).length;
+                // Ensure that this is an independent set of accessors
+                if (equalCount === accessorsLength && indicesAccessor === primitiveIndices) {
+                    return primitive;
+                }
+            }
+        }
+    }
+}
+
+function cacheOptimization(gltf, cacheSize) {
+    // perform post vertex cache optimization
+    var primitive;
+    var geometry;
+    var meshes = gltf.meshes;
+    for (var meshId in meshes) {
+        if (meshes.hasOwnProperty(meshId)) {
+            var mesh = meshes[meshId];
+            var primitives = mesh.primitives;
+            var primitivesLength = primitives.length;
+            for (var i = 0; i < primitivesLength; i++) {
+                primitive = primitives[i];
+                geometry = gltfPrimitiveToCesiumGeometry(gltf, primitive);
+                GeometryPipeline.reorderForPostVertexCache(geometry, cacheSize);
+
+                cesiumGeometryToGltfPrimitive(gltf, primitive, geometry);
+            }
+        }
+    }
+    // perform pre vertex cache optimization on each independent primitive
+    var usageTables = createAccessorUsageTables(gltf);
+    // Iterate through each usage table
+    for (var j = 0; j < usageTables.length; j++) {
+        var indexAccessors = usageTables[j].indexAccessors;
+        var attributeAccessors = usageTables[j].attributeAccessors;
+        var keys = Object.keys(indexAccessors);
+        // If a set of attributes accessors is only used by one indices accessor
+        if (keys.length === 1) {
+            var indicesId = keys[0];
+            // look through to see if an independent primitive exists for those attributes
+            primitive = getAccessorsPrimitive(gltf, attributeAccessors, indicesId);
```

Comment:
Doing this search every time is really inefficient. Iterate over the primitives and put them into a dicitionary of arrays where the keys are the index accessor ids before you go through the usage tables. Then here, you can just look up the primitives in ~O(1).